src/PHPUnit/Util/Log/JSON.php

```diff
@@ -30,17 +42,17 @@ class JSON extends \PHPUnit\Util\Printer implements \PHPUnit\Framework\TestListe
     /**
      * An error occurred.
      *
-     * @param \PHPUnit\Framework\Test $test
-     * @param \Throwable $e
+     * @param TestCase|string $test
```

Comment:
Where did the string come from?
Nobody has every raised such issue, I think that this change is mistake.

Reply:
In PHPUnit 8/9 the interfaces inside `src/Runner/Hook/*.php` type these parameters as strings, I found it too strange to completely overwrite that signature. Do I leave it as it was before?

Reply:
I think that this is related to TestListener deprecation in PHPUnit 8/9.
Codeception hasn't migrated from TestListener to TestHooks.
TestListener is no longer deprecated in PHPUnit 10, and TestHooks are deprecated now: https://github.com/sebastianbergmann/phpunit/issues/4595
There is no need to support them.

Reply:
@Naktibalda OK, I wasn't aware of that. Thank you. I re-implemented the TestListener interface and made the corresponding changes.